src/nrnpython/hoccontext.h

```diff
@@ -37,3 +25,22 @@ static void hc_restore_(HocContext* hc) {
     hoc_objectdata = hc->obd;
     hoc_symlist = hc->sl;
 }
+
+class HocTopContext {
```

Comment:
I think would be nice to emphasize in the name that this is `RAII` `scoped` class. Otherwise the declaration looks like a dead variable.

Reply:
In Python these are basically context managers, so maybe `HocTopContextManager`?

Reply:
fine for me